### PR TITLE
spg Chapter 14

### PR DESCRIPTION
**Pull request by SimonGross**
Chapter 14

### CODE REVIEW

File: src/com/datacash/training/chapter14/answers.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+14) `MyClass::myStaticmethod`
+
+15) `mcObj::myInstMethod`
+
+16) Question16.java
+
+17) `Class::new`, where `Class` is the class whose constructor is being referenced.
+
+18) java.utils.function
```
Comment (codavex) [inline src/com/datacash/training/chapter14/answers.md:40]: util, not utils, which would get flagged up straight away anyway. Other than that everything else looks good :+1:

File: src/com/datacash/training/chapter14/Question9.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	 
+	  public static void main(String args[]) 
+	  { 
+	    String inStr = "Lambda Expressions Expand Java"; 
+	    String outStr; 
+	 
+	    System.out.println("Input string: " + inStr); 
+	 
+	    // Remove spaces
+	    StringFunc noSpaces = (str) -> str.replace(" ", "");
```
Comment (codavex) [inline src/com/datacash/training/chapter14/Question9.java:21]: Neater than the books solution of iterating over the length of the string

File: src/com/datacash/training/chapter14/Question16.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+	// Return true if n is a factor of i.
+	private boolean isFactor( int n, int i ) {
+		return (i % n) == 0;
+	}
+
+	public boolean hasCommonFactor( int i ) {
+
+		for ( int n = 2 ; n < v / n ; n++ ) {
+			if ( isFactor( n ) && isFactor( n, i ) ) {
```
Comment (codavex) [inline src/com/datacash/training/chapter14/Question16.java:34]: Again, much nicer than the books solution. He doesn't bother to use the predefined isFactor

File: src/com/datacash/training/chapter14/answers.md (hunk 1 above)
Comment (SandyThomson) [inline src/com/datacash/training/chapter14/answers.md:40]: Doh ... Ballsed that up.

File: src/com/datacash/training/chapter14/Question9.java (hunk 2 above)
Comment (SandyThomson) [inline src/com/datacash/training/chapter14/Question9.java:21]: Yeah, you'd be insane to write it like that (the way the book does it) .... I'm starting to think the author is insane.

**Pull request closed without merging**